src-tauri/src/graph.rs

```diff
@@ -0,0 +1,317 @@
+use std::collections::{HashMap, HashSet};
+
+#[derive(Debug)]
```

Comment:
Is this something we need to keep in the codebase?

Reply:
We need this to see the output of print statements in tests. I think it's best to keep them for now and I can remove later.